code/test/util.js

```diff
@@ -215,6 +269,123 @@ const getEOTP = async ({ seed, hseed, effectiveTime, timeOffset }) => {
   return { index, eotp }
 }
 
+// transactionExecute commits and reveals a wallet transaction
+const transactionExecute = async ({ wallet, operationType, tokenType, contractAddress, tokenId, dest, amount, data, address, randomSeed, backlinkAddresses, testTime }) => {
+  if (testTime === undefined) { testTime = Date.now() }
+  // // calculate counter from testTime
+  const counter = Math.floor(testTime / INTERVAL)
+  const otp = ONEUtil.genOTP({ seed: wallet.seed, counter })
+  // // calculate wallets effectiveTime (creation time) from t0
+  const info = await wallet.wallet.getInfo()
+  const t0 = new BN(info[3]).toNumber()
```

Comment:
t0 is in wallet

Reply:
These two lines are pulling t0 from the wallet
```
  const t0 = new BN(info[3]).toNumber()
  const walletEffectiveTime = t0 * INTERVAL
```
Do you have a recommendation on a more effective/cleaner way to do this?

Reply:
it is in the walletInfo object you passed in